### ISSUE TITLE
Domain squatting

### ISSUE THREAD

**Issue by intraserv** [dot-src/dot-src#9]
Now this is a big issue for me as i have had to pay in excess of £100 for my personal domain name as it was being squatted by HugeDomains - I think that there should be a NO-RESELL policy or something like that, meaning that you can't buy a domain solely for the purpose of re-selling it. Don't know how you'd check it though, Elaborate code or Human Labor?
anyway definitely +1 for .src - would be amazing for my projects!

**Comment by joallard** [comment]
What about no-resell, and expiring domains go into a one-week auction after expiring?

**Comment by InukVT** [comment]
How about one month auction? Well, that is technical stuff that needs to be done after gTLD registrations..

**Comment by nirix** [comment]
No-resell is a must. We don't want people to come along, buy up `jquery.src` for example and try to sell it for an insane amount.

Also, if a domain were to expire, we could allow 30 days for the original owner to renew it; after the 30 days it could go to auction or simply be registered by someone else.

**Comment by joallard** [comment]
We could stipulate that in spirit, no squatting, and that the registry will make a best effort to assign domain names to projects generally-known-as that name. Ie. if you register `jquery.src`, I'm sorry, but we're going to give that name to the actual jQuery project, that is "recognized as being jQuery".

Also, you can't hold a domain and do nothing with it. A domain needs to host a reasonable bona-fide project. ~~Say you have one week to conform to that after registering.~~ Wait, no actually, if you need a domain, you need code from day 1.

Let's see. If I'm bad-man #1, and I try to register `jquery.src`, won't work, because everyone knows what jQuery is. Let's say I register a more obscure project that already exists, [`trasto.src`](//github.com/yabawock/trasto). Can't do that, even if I put some piece of code, cause even if it's not known, it's been around for kind of a while and has some kind of old codebase.

Let's say I register `supergoodidea.src`, and I don't follow through with the project. And another one comes along and becomes more successful; I guess we hand it over to the new project?

I just don't see what the auction is for. Maybe some inactivity/transfer scenarios? Eg. Someone registers `bloop.src`, makes a small project, then it's abandoned and years later another bigger `bloop` comes along. (How would we deal with this situation btw?)

Some really drafty policies:

> - The src registry has for mission to provide a domain namespace to host open-source code projects.
> - The interest of the community of finding projects easily and at their expected location primes over the interest of the individuals.
> - A project "generally known as" a certain name should ideally reside at that name in the src namespace. 
> - A domain that is occupied by a smaller project can be claimed by a more-established project. Priority will be given to names that are in the mindshare of people.
> - A domain needs to serve a purpose of hosting a project. No domain can be reserved for only the purpose of reserving it.
> - For our purposes, a project is a repository of open-source code that is non-trivial, original and useful for the community.
> - All domains are property of the registry and cannot be sold.

**Comment by ghost** [comment]
In my opinion, legitimate users who just keep the domain but don't have a specific idea for it should have their domain kept theirs.

Users who may abuse this to resell the domain are banned.

**Comment by noahcoetsee** [comment]
> Also, you can't hold a domain and do nothing with it. A domain needs to host a reasonable bona-fide project. Say you have one week to conform to that after registering. Wait, no actually, if you need a domain, you need code from day 1.

Joallard, that is noteable, but you must have reasonable guidelines and rules that actually define what a project is, or some people (who must really want to hold on to their domain) might take it to court. And about the 1 day code rule. Some people may actually have future plans, and want to make sure they have a safe domain for their project. Their sole intent may be to have a nice little .src domain for their KittyJS project. But it's not near being finished, but they did see that there was another project called Kitty.js out there, and they wanted to make sure they got that domain, because they knew they would need an awesome Javascript hacker-run TLD to attract the awesome attention it would already be getting!

**Against**
> A domain needs to serve a purpose of hosting a project. No domain can be reserved for only the purpose of reserving it.

**For**
> In my opinion, legitimate users who just keep the domain but don't have a specific idea for it should have their domain kept theirs.

This is true, someone may purchase it, and not know what to put on the actual domain, besides redirect to a readme or github. This should be acceptable. Once again, pushing against joallards 1 day code rule. Even pushing against an any-day code rule.

> Users who may abuse this to resell the domain are banned.

Maybe. It is more likely that people will just begin selling the actual registry emails and passwords, and it goes on forever. Even if you have an official resell, one could still bypass that and sell the account used to create the domain. Abusing the resell may be punishable, but once again, you must have rules and/or guidelines that keep the lawyers/judges off our backs! So basically, you better have a good system of finding those that abuse, and definitely encourage users to report such behavior.

- Noah

**Comment by joallard** [comment]
> Their sole intent may be to have a nice little .src domain for their KittyJS project. But it's not near being finished, but they did see that there was another project called Kitty.js out there, and they wanted to make sure they got that domain

See? If there's already a Kitty.js out there, you shouldn't be allowed to reserve it if you don't show code for it. The question is open if you should be able to register it on your own at all.

> In my opinion, legitimate users who just keep the domain but don't have a specific idea for it should have their domain kept theirs.

> This is true, someone may purchase it, and not know what to put on the actual domain, besides redirect to a readme or github. This should be acceptable. 

My opinion is firmly against this. Namespace exhaustion is real. And not getting the right names to the right projects runs counter to the project's very spirit.

> Users who may abuse this to resell the domain are banned.

Yep, that sounds difficult, because how do you (a) define, and (b) catch abuse? Those are doable, but hard problems I'd rather not have the registry deal with. Just like software, terms are easier to run and debug the simpler they are.
